### DISCUSSION TITLE
Using a drag-and-drop hook ref

### DISCUSSION THREAD

**Discussion by helt** [KevinVandy/material-react-table#984] [Q&A]
Is there a way to set a ref onto the drag-handle, so I can use the ref I got from my dnd-hooks? (react-dnd in this case)

I have a project which uses react-dnd for drag and drop everywhere. However, for material-react-table, I had to use custom code which works with the dragevent dataTransfer directly, because adding a ref to the `muiRowDragHandleProps` was forbidden. 

Is it an option to *just* replace the [<MRT_TableBodyRowGrabHandle />](https://www.material-react-table.com/docs/api/mrt-components#%3Cmrt_tablebodyrowgrabhandle--%3E) by my own one, which also includes a ref, or is that not possible (yet)?

**Comment by KevinVandy** [comment]
react-dnd usually breaks MRT's dragging implementation.
